devicetypes/axis/axis-gear-st.src/axis-gear-st.groovy

```diff
@@ -316,33 +314,44 @@ def configure() {
 						zigbee.readAttribute(CLUSTER_ONOFF, ONOFF_ATTR_ONOFFSTATE) +
 						zigbee.readAttribute(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL) +
 						zigbee.readAttribute(CLUSTER_POWER, POWER_ATTR_BATTERY)
-						
-	def cmds = zigbee.configureReporting(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_ATTR_LIFTPERCENTAGE, 0x20, 1, 3600, 0x00) + 
+
+	def cmds = zigbee.configureReporting(CLUSTER_WINDOWCOVERING, WINDOWCOVERING_ATTR_LIFTPERCENTAGE, 0x20, 1, 3600, 0x00) +
 				zigbee.configureReporting(CLUSTER_ONOFF, ONOFF_ATTR_ONOFFSTATE, 0x10, 1, 3600, 0x00) +
 				zigbee.configureReporting(CLUSTER_LEVEL, LEVEL_ATTR_LEVEL, 0x20, 1, 3600, 0x00) +
 				zigbee.configureReporting(CLUSTER_POWER, POWER_ATTR_BATTERY, 0x20, 1, 3600, 0x01)
-				
+
 	log.info "configure() --- cmds: $cmds"
 	return attrs_refresh + cmds
 }
 
 def parse(String description) {
 	log.trace "parse() --- description: $description"
-	
+
 	Map map = [:]
 
+	if (device.currentValue("shadeLevel") == null && device.currentValue("level") != null) {
+		sendEvent(name: "shadeLevel", value: device.currentValue("level"), unit: "%")
+	}
+
 	def event = zigbee.getEvent(description)
 	if (event && description?.startsWith('on/off')) {
 		log.trace "sendEvent(event)"
 		sendEvent(event)
 	}
-	
+
 	else if ((description?.startsWith('read attr -')) || (description?.startsWith('attr report -'))) {
 		map = parseReportAttributeMessage(description)
 		def result = map ? createEvent(map) : null
+
+		if (map.name == "level") {
+			def shadeLevelMap = map.findAll { it.key != "displayed" }
+			shadeLevelMap.name = "shadeLevel"
```

Comment:
I'm very confused about what's going on here, can you explain @dkirker ?

Reply:
This is just creating a shadeLevel event if `parseReportAttributeMessage` returned a level event, but excluding the displayed property (as we don't want to force it to false).

Reply:
why not just `map.remove("displayed")`?

Reply:
That'd probably work, too. However, I just opted to go with something even more simpler.